pkg/auth/auth.go

```diff
@@ -79,20 +80,20 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		if !opts.AcceptUnspecifiedRegistry {
 			return errors.New("please provide a registry to login to")
 		}
-		if server, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
+		if key, err = defaultRegistryWhenUnspecified(systemContext); err != nil {
 			return err
 		}
-		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
-	} else {
-		server = getRegistryName(args[0])
+		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", key)
 	}
```

Comment:
⚠️ `key` is, AFAICS, unset if `len(args) == 1`

Reply:
> ⚠️ `key` is, AFAICS, unset if `len(args) == 1`

[Are there any tests, maybe in Podman, that break on such problems? If not, where to add them?]

Reply:
Fixed :+1: